tangram-rmrk/src/lib.rs

```diff
@@ -0,0 +1,108 @@
+// TODO: Tangram storage
+// TODO: Tangram functions: do_mint_nft, do_create_collection, do_create_realm
+
+
+		/// Mints an NFT in the specified collection
+		/// Sets metadata and the royalty attribute
+		///
+		/// Parameters:
+		/// - `collection_id`: The class of the asset to be minted.
+		/// - `nft_id`: The nft value of the asset to be minted.
+		/// - `recipient`: Receiver of the royalty
+		/// - `royalty`: Permillage reward from each trade for the Recipient
+		/// - `metadata`: Arbitrary data about an nft, e.g. IPFS hash
+		#[pallet::weight(10_000 + T::DbWeight::get().reads_writes(1,1))]
+		#[transactional]
+		pub fn mint_nft(
+			origin: OriginFor<T>,
+			owner: T::AccountId,
+			collection_id: CollectionId,
+			recipient: Option<T::AccountId>,
+			royalty: Option<Permill>,
+			metadata: BoundedVec<u8, T::StringLimit>,
+		) -> DispatchResult {
+
+			let sender = ensure_signed(origin.clone())?;
+			if let Some(collection_issuer) =
+				pallet_uniques::Pallet::<T>::class_owner(&collection_id)
+			{
+				ensure!(collection_issuer == sender, Error::<T>::NoPermission);
+			} else {
+				return Err(Error::<T>::CollectionUnknown.into())
+			}
+
```

Comment:
maybe also worth to compare with orml nft

Reply:
Yes, we'll check it. This solution seems suboptimal, I don't like the complexity of having three storage layers. This will impact weights significantly.

Reply:
Another way - use a wrapper around Uniques pallet, like RMRK did.

Reply:
if we want rmrk compat then the way is using rmrk and carrying another map for realms.
if we brew our own, we just wrap uniques with all the stuff from tangram, possibly not much different to rmrk at the end, as the features are kind of logical if you want stuff like composability, mutation, royalties